extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -426,6 +426,16 @@ static <DATA_TYPE> void writeColumnSource(
         LongSupplier rowStepGetter;
         LongSupplier valuesStepGetter;
 
+        // These two lists are parallel, where each element represents a single page. The rawItemCountPerPage
+        // contains the number of items (the sum of the array sizes for each containing row) contained within the
+        // page.
+        final TIntArrayList rawItemCountPerPage = new TIntArrayList();
+
+        // The originalRowsPerPage contains the number of arrays (rows) from the original table contained within
+        // the page.
+        final TIntArrayList originalRowsPerPage = new TIntArrayList();
```

Comment:
Your comment is explanatory and helpful. The lists themselves could be allocated only in the case where they are used, and assigned to `null` in the other case, which also lets you eliminate the `case1` boolean.

I think the names `rowStepGetter`, `valuesStepGetter`, `rawItemCountPerPage`, and `originalRowsPerPage` could be improved; your comment goes a long way towards improving the internal documentation in this code, however.

One approach we could take would be to clean up the whole mess a bit further by introducing some kind of structure around the two cases. It would be nice to have a way to say that having `lengthSource`, a `rowSet` that isn't the `tableRowSet`/`originalRowSet`, the two lists, and the length supplier could be all or none. That would also give us a place to centralize the documentation, instead of having it splayed across three methods, with variable naming differences between the caller and callees, etc.